### PR TITLE
Fixed Tier input on sponsors page

### PR DESCRIPTION
**Pull request by Lewin-B**
Added a selection menu to the tier input in the sponsors page to avoid type errors being logged to the console. Also formatted and cleaned up some code in the sponsors page.

### CODE REVIEW

File: apps/admin/src/pages/sponsors/add-sponsor-form.tsx [hunk 1]
```diff
@@ -53,9 +53,11 @@ export function AddSponsorForm() {
   });
 
   // Fill with valid hackathonId's
-  // Use hook directly within the component body or useEffect
   const { data: hackathons } = trpc.hackathons.getAll.useQuery();
 
+  // initialize tiers
+  const tiers = ["gold", "silver", "bronze"];
```
Comment (LightBounded) [inline apps/admin/src/pages/sponsors/add-sponsor-form.tsx]: This constant should probably be moved to `packages/db/src/consts` and then imported accordingly.

File: apps/admin/src/pages/sponsors/add-sponsor-form.tsx [hunk 2]
```diff
@@ -120,9 +122,20 @@ export function AddSponsorForm() {
...
-                <Input placeholder="Tier" {...field} />
-              </FormControl>
+              <Select onValueChange={field.onChange}>
+                <FormControl>
+                  <SelectTrigger>
+                    <SelectValue placeholder="Select Tier" />
+                  </SelectTrigger>
+                </FormControl>
+                <SelectContent>
+                  {tiers.map((Tier) => (
```
Comment (LightBounded) [inline apps/admin/src/pages/sponsors/add-sponsor-form.tsx]: Nit: but callback arguments like `Tier`, in this case, should be camelCased.

File: apps/admin/src/pages/sponsors/update-sponsor-form.tsx [hunk 3]
```diff
@@ -69,6 +69,9 @@ export function UpdateSponsorForm({ sponsor }: { sponsor: Sponsor }) {
   // fill array with all valid hackathonIds
   const { data: hackathons } = trpc.hackathons.getAll.useQuery();
 
+  // Initialize tiers array
+  const tiers = ["gold", "silver", "bronze"];
```
Comment (LightBounded) [inline apps/admin/src/pages/sponsors/update-sponsor-form.tsx]: See my first comment.

File: apps/admin/src/pages/sponsors/update-sponsor-form.tsx [hunk 4]
```diff
@@ -129,9 +132,20 @@ export function UpdateSponsorForm({ sponsor }: { sponsor: Sponsor }) {
...
-                <Input placeholder="tier" {...field} />
-              </FormControl>
+              <Select onValueChange={field.onChange} defaultValue={field.value}>
+                <FormControl>
+                  <SelectTrigger>
+                    <SelectValue placeholder="Select Tier" />
+                  </SelectTrigger>
+                </FormControl>
+                <SelectContent>
+                  {tiers.map((Tier) => (
```
Comment (LightBounded) [inline apps/admin/src/pages/sponsors/update-sponsor-form.tsx]: See my comment about callback arguments

**Review by LightBounded — CHANGES_REQUESTED** [review]
Left some comments. Otherwise looks good!

File: apps/admin/src/pages/sponsors/add-sponsor-form.tsx (hunk 1 above)
Comment (LightBounded) [reply apps/admin/src/pages/sponsors/add-sponsor-form.tsx]: Constants look like this: 
```ts
export const gradYears = [
  "2024",
  "2025",
  "2026",
  "2027",
  "2028",
  "other",
] as const;
```
Notice the `as const` which ensures this object is immutable.

Comment (LightBounded) [reply apps/admin/src/pages/sponsors/add-sponsor-form.tsx]:
```ts

export const sponsors = sqliteTable("sponsors", {
  id: integer("id").primaryKey({ autoIncrement: true }),
  name: text("name").notNull(),
  logo: text("logo").notNull(),
  website: text("website").notNull(),
  tier: text("tier", { enum: ["gold", "silver", "bronze", "other"] }).notNull(),
  hackathonId: integer("hackathon_id").references(() => hackathons.id, {
    onDelete: "cascade", // If the hackathon is deleted, delete the hacker
    onUpdate: "cascade", // If the hackathon is updated, update the hacker
  }),
});
```

Another thing is in `packages/db/src/schemas/schema.ts`, which is where the above code snippet comes from should be updated to where the enum property should point to the tiers constant.

File: apps/admin/src/pages/sponsors/add-sponsor-form.tsx (hunk 2 above)
Comment (Lewin-B) [reply apps/admin/src/pages/sponsors/add-sponsor-form.tsx]: Understood

File: apps/admin/src/pages/sponsors/update-sponsor-form.tsx (hunk 3 above)
Comment (Lewin-B) [reply apps/admin/src/pages/sponsors/update-sponsor-form.tsx]: That was actually your second comment big bro (i lied)

File: apps/admin/src/pages/sponsors/update-sponsor-form.tsx (hunk 4 above)
Comment (Lewin-B) [reply apps/admin/src/pages/sponsors/update-sponsor-form.tsx]: Yes sir

File: apps/admin/src/pages/sponsors/add-sponsor-form.tsx (hunk 1 above)
Comment (Lewin-B) [reply apps/admin/src/pages/sponsors/add-sponsor-form.tsx]: What a marvelous convention. Implementing this fills me with excitement.

**Comment by LightBounded** [comment]
You are failing CI on formatting btw

**Review by LightBounded — APPROVED** [review]
lgtm

**Pull request merged**